File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/validate_agent_flow_diagram_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ValidateAgentFlowDiagramDetails(object):
    """
    Request payload for validating an Agent Flow diagram without persisting changes.
    """

    #: A constant which can be used with the validation_level property of a ValidateAgentFlowDiagramDetails.
    #: This constant has a value of "BASIC"
    VALIDATION_LEVEL_BASIC = "BASIC"

    #: A constant which can be used with the validation_level property of a ValidateAgentFlowDiagramDetails.
    #: This constant has a value of "DEEP"
    VALIDATION_LEVEL_DEEP = "DEEP"

    def __init__(self, **kwargs):
        """
        Initializes a new ValidateAgentFlowDiagramDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param agent_flow_key:
            The value to assign to the agent_flow_key property of this ValidateAgentFlowDiagramDetails.
        :type agent_flow_key: str

        :param agent_flow_diagram:
            The value to assign to the agent_flow_diagram property of this ValidateAgentFlowDiagramDetails.
        :type agent_flow_diagram: oci.aidataplatform_dp.models.AgentFlowDiagram

        :param model_version:
            The value to assign to the model_version property of this ValidateAgentFlowDiagramDetails.
        :type model_version: str

        :param validation_level:
            The value to assign to the validation_level property of this ValidateAgentFlowDiagramDetails.
            Allowed values for this property are: "BASIC", "DEEP"
        :type validation_level: str

        :param options:
            The value to assign to the options property of this ValidateAgentFlowDiagramDetails.
        :type options: oci.aidataplatform_dp.models.AgentFlowValidationOptions

        """
        self.swagger_types = {
            'agent_flow_key': 'str',
            'agent_flow_diagram': 'AgentFlowDiagram',
            'model_version': 'str',
            'validation_level': 'str',
            'options': 'AgentFlowValidationOptions'
        }

        self.attribute_map = {
            'agent_flow_key': 'agentFlowKey',
            'agent_flow_diagram': 'agentFlowDiagram',
            'model_version': 'modelVersion',
            'validation_level': 'validationLevel',
            'options': 'options'
        }

        self._agent_flow_key = None
        self._agent_flow_diagram = None
        self._model_version = None
        self._validation_level = None
        self._options = None

    @property
    def agent_flow_key(self):
        """
        Gets the agent_flow_key of this ValidateAgentFlowDiagramDetails.
        Optional Agent Flow key used to resolve persisted validation context.


        :return: The agent_flow_key of this ValidateAgentFlowDiagramDetails.
        :rtype: str
        """
        return self._agent_flow_key

    @agent_flow_key.setter
    def agent_flow_key(self, agent_flow_key):
        """
        Sets the agent_flow_key of this ValidateAgentFlowDiagramDetails.
        Optional Agent Flow key used to resolve persisted validation context.


        :param agent_flow_key: The agent_flow_key of this ValidateAgentFlowDiagramDetails.
        :type: str
        """
        self._agent_flow_key = agent_flow_key

    @property
    def agent_flow_diagram(self):
        """
        **[Required]** Gets the agent_flow_diagram of this ValidateAgentFlowDiagramDetails.

        :return: The agent_flow_diagram of this ValidateAgentFlowDiagramDetails.
        :rtype: oci.aidataplatform_dp.models.AgentFlowDiagram
        """
        return self._agent_flow_diagram

    @agent_flow_diagram.setter
    def agent_flow_diagram(self, agent_flow_diagram):
        """
        Sets the agent_flow_diagram of this ValidateAgentFlowDiagramDetails.

        :param agent_flow_diagram: The agent_flow_diagram of this ValidateAgentFlowDiagramDetails.
        :type: oci.aidataplatform_dp.models.AgentFlowDiagram
        """
        self._agent_flow_diagram = agent_flow_diagram

    @property
    def model_version(self):
        """
        Gets the model_version of this ValidateAgentFlowDiagramDetails.
        The model and upgrade compatibility version for this validation request.


        :return: The model_version of this ValidateAgentFlowDiagramDetails.
        :rtype: str
        """
        return self._model_version

    @model_version.setter
    def model_version(self, model_version):
        """
        Sets the model_version of this ValidateAgentFlowDiagramDetails.
        The model and upgrade compatibility version for this validation request.


        :param model_version: The model_version of this ValidateAgentFlowDiagramDetails.
        :type: str
        """
        self._model_version = model_version

    @property
    def validation_level(self):
        """
        Gets the validation_level of this ValidateAgentFlowDiagramDetails.
        Validation depth to apply.

        Allowed values for this property are: "BASIC", "DEEP"


        :return: The validation_level of this ValidateAgentFlowDiagramDetails.
        :rtype: str
        """
        return self._validation_level

    @validation_level.setter
    def validation_level(self, validation_level):
        """
        Sets the validation_level of this ValidateAgentFlowDiagramDetails.
        Validation depth to apply.


        :param validation_level: The validation_level of this ValidateAgentFlowDiagramDetails.
        :type: str
        """
        allowed_values = ["BASIC", "DEEP"]
        if not value_allowed_none_or_none_sentinel(validation_level, allowed_values):
            raise ValueError(
                "Invalid value for `validation_level`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._validation_level = validation_level

    @property
    def options(self):
        """
        Gets the options of this ValidateAgentFlowDiagramDetails.

        :return: The options of this ValidateAgentFlowDiagramDetails.
        :rtype: oci.aidataplatform_dp.models.AgentFlowValidationOptions
        """
        return self._options

    @options.setter
    def options(self, options):
        """
        Sets the options of this ValidateAgentFlowDiagramDetails.

        :param options: The options of this ValidateAgentFlowDiagramDetails.
        :type: oci.aidataplatform_dp.models.AgentFlowValidationOptions
        """
        self._options = options

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/ValidateAgentFlowDiagramDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Request payload for validating an Agent Flow diagram without persisting changes.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ValidateAgentFlowDiagramDetails.Builder.class)

public final class ValidateAgentFlowDiagramDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"agentFlowKey", "agentFlowDiagram", "modelVersion", "validationLevel", "options"})
    public ValidateAgentFlowDiagramDetails(String agentFlowKey, AgentFlowDiagram agentFlowDiagram, String modelVersion, ValidationLevel validationLevel, AgentFlowValidationOptions options) {
        super();
        this.agentFlowKey = agentFlowKey;
        this.agentFlowDiagram = agentFlowDiagram;
        this.modelVersion = modelVersion;
        this.validationLevel = validationLevel;
        this.options = options;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Optional Agent Flow key used to resolve persisted validation context.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
private String agentFlowKey;

        /**
         * Optional Agent Flow key used to resolve persisted validation context.
         * @param agentFlowKey the value to set
         * @return this builder
         **/
        

public Builder agentFlowKey(String agentFlowKey) {
    this.agentFlowKey = agentFlowKey;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("agentFlowDiagram")
private AgentFlowDiagram agentFlowDiagram;



public Builder agentFlowDiagram(AgentFlowDiagram agentFlowDiagram) {
    this.agentFlowDiagram = agentFlowDiagram;
    return this;
}
            /**
     * The model and upgrade compatibility version for this validation request.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("modelVersion")
private String modelVersion;

        /**
         * The model and upgrade compatibility version for this validation request.
         * @param modelVersion the value to set
         * @return this builder
         **/
        

public Builder modelVersion(String modelVersion) {
    this.modelVersion = modelVersion;
    return this;
}
            /**
     * Validation depth to apply.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("validationLevel")
private ValidationLevel validationLevel;

        /**
         * Validation depth to apply.
         * @param validationLevel the value to set
         * @return this builder
         **/
        

public Builder validationLevel(ValidationLevel validationLevel) {
    this.validationLevel = validationLevel;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("options")
private AgentFlowValidationOptions options;



public Builder options(AgentFlowValidationOptions options) {
    this.options = options;
    return this;
}


        public ValidateAgentFlowDiagramDetails build() {
            ValidateAgentFlowDiagramDetails model = new ValidateAgentFlowDiagramDetails(this.agentFlowKey
                , this.agentFlowDiagram
                , this.modelVersion
                , this.validationLevel
                , this.options);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ValidateAgentFlowDiagramDetails model) {
                this.agentFlowKey(model.getAgentFlowKey());
    this.agentFlowDiagram(model.getAgentFlowDiagram());
    this.modelVersion(model.getModelVersion());
    this.validationLevel(model.getValidationLevel());
    this.options(model.getOptions());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Optional Agent Flow key used to resolve persisted validation context.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
    private final String agentFlowKey;

        /**
     * Optional Agent Flow key used to resolve persisted validation context.
     * @return the value
     **/
    
    public String getAgentFlowKey() {
        return agentFlowKey;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowDiagram")
    private final AgentFlowDiagram agentFlowDiagram;

    
    public AgentFlowDiagram getAgentFlowDiagram() {
        return agentFlowDiagram;
    }


        /**
     * The model and upgrade compatibility version for this validation request.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("modelVersion")
    private final String modelVersion;

        /**
     * The model and upgrade compatibility version for this validation request.
     * @return the value
     **/
    
    public String getModelVersion() {
        return modelVersion;
    }

    /**
     * Validation depth to apply.
     **/
    public enum ValidationLevel implements com.oracle.bmc.http.internal.BmcEnum {
        Basic("BASIC"),
        Deep("DEEP"),
        ;

        

        private final String value;
        private static java.util.Map<String, ValidationLevel> map;

        static {
            map = new java.util.HashMap<>();
            for (ValidationLevel v : ValidationLevel.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        ValidationLevel(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static ValidationLevel create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid ValidationLevel: " + key);
        }
    };
        /**
     * Validation depth to apply.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("validationLevel")
    private final ValidationLevel validationLevel;

        /**
     * Validation depth to apply.
     * @return the value
     **/
    
    public ValidationLevel getValidationLevel() {
        return validationLevel;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("options")
    private final AgentFlowValidationOptions options;

    
    public AgentFlowValidationOptions getOptions() {
        return options;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ValidateAgentFlowDiagramDetails(");
        sb.append("agentFlowKey=").append(String.valueOf(this.agentFlowKey));
        sb.append(", agentFlowDiagram=").append(String.valueOf(this.agentFlowDiagram));
        sb.append(", modelVersion=").append(String.valueOf(this.modelVersion));
        sb.append(", validationLevel=").append(String.valueOf(this.validationLevel));
        sb.append(", options=").append(String.valueOf(this.options));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ValidateAgentFlowDiagramDetails)) {
            return false;
        }

        ValidateAgentFlowDiagramDetails other = (ValidateAgentFlowDiagramDetails) o;
        return java.util.Objects.equals(this.agentFlowKey, other.agentFlowKey) &&
            java.util.Objects.equals(this.agentFlowDiagram, other.agentFlowDiagram) &&
            java.util.Objects.equals(this.modelVersion, other.modelVersion) &&
            java.util.Objects.equals(this.validationLevel, other.validationLevel) &&
            java.util.Objects.equals(this.options, other.options);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.agentFlowKey == null ? 43 : this.agentFlowKey.hashCode());
        result = (result * PRIME) + (this.agentFlowDiagram == null ? 43 : this.agentFlowDiagram.hashCode());
        result = (result * PRIME) + (this.modelVersion == null ? 43 : this.modelVersion.hashCode());
        result = (result * PRIME) + (this.validationLevel == null ? 43 : this.validationLevel.hashCode());
        result = (result * PRIME) + (this.options == null ? 43 : this.options.hashCode());
        return result;
    }


}
